### DISCUSSION TITLE
Replace Gluu branding with Janssen Branding for User Facing Pages (Login, Error, etc)

### DISCUSSION THREAD

**Discussion by ossdhaval** [JanssenProject/jans#7117] [General]
I have started to implement the code changes for [this redesign](https://github.com/JanssenProject/jans/issues/1302). A good amount of CSS and HTML would be changed to implement this. 

JS libraries and CSS that are used in current pages seem out of date now. This is probably the right time to upgrade them. The bootstrap library that we are using is version-4. That was Released 5 years back. The latest is 5.3.2. And we have a bunch of other libraries. For instance, we are currently using fontawesome and jQuery too. As I read, it seems that if we upgrade to Bootstrap 5, we don't need the previous two libraries either.

But I am no SME in this area. So, I am putting this thought out for discussion. Should we upgrade to Bootstrap 5 and remove fontawesome and jQuery? 

Thoughts? 

CC: @duttarnab @moabu @nynymike

**Comment by BainMcKay** [comment]
On a priority basis, we need an Admin UI and we need an export/import  from GLUU 4 to Jansen., so there is a one-to-one mapping.  A single click migration would save a ton of time, and frustration working through ambiguous documentation where details are understated or assumed.

**Reply by ossdhaval** [reply]
@BainMcKay 

This thread is about changes to web pages (login, device-code etc) to give them `Janssen` identity. We can discuss above concerns in a different discussion thread.

Re - admin UI - From the Janssen Project perspective, we have TUI to ease the administration. And we are building more and more features into it. Proposals for additional features are welcome :-)

**Reply by nynymike** [reply]
Dhaval is correct. This thread is about replacing user-facing login pages that might still have the Gluu trademark.

**Comment by jv18creator** [comment]
Here are my thoughts from some research, do correct me where I might be wrong,

> As I read, it seems that if we upgrade to Bootstrap 5, we don't need the previous two libraries either.

I belive we could remove jQuery-ui library, jQuery is used for DOM manipulation which is used in the project & Upgrading Bootstrap alone won't replace the core functionality provided by jQuery.

> Should we upgrade to Bootstrap 5 and remove fontawesome and jQuery?

- I searched for the usage of jQuery & it's used in [login-template.xhtml](https://github.com/JanssenProject/jans/blob/a134ac9fd4626f7ad5ee24a6680beaa7f094a056/jans-auth-server/server/src/main/webapp/WEB-INF/incl/layout/login-template.xhtml#L24) & [login-extended-template.xhtml](https://github.com/JanssenProject/jans/blob/a134ac9fd4626f7ad5ee24a6680beaa7f094a056/jans-auth-server/server/src/main/webapp/WEB-INF/incl/layout/login-extended-template.xhtml#L30) 
- We won't be able to remove fontawesome library as well because from the [bootstrap 5 docs](https://getbootstrap.com/docs/5.0/extend/icons/) they are supporting [bootstrap-icons](https://icons.getbootstrap.com/) which is not included in v5 of bootstrap and needs to be separately installed just like fontawesome.
- In conclusion, we could remove the [jQuery-ui v1.12.1](https://jqueryui.com/download/#!version=1.12.1) library & test it because it's only used at [login-extended-template.xhtml](https://github.com/JanssenProject/jans/blob/a134ac9fd4626f7ad5ee24a6680beaa7f094a056/jans-auth-server/server/src/main/webapp/WEB-INF/incl/layout/login-extended-template.xhtml#L46), upgrade bootstrap to version 5 & [fontawesome to v6](https://fontawesome.com/docs/web/setup/get-started) & keep jQuery.

**Reply by nynymike** [reply]
Very good suggestions. Please proceed.

**Reply by jv18creator** [reply]
thanks @nynymike, i will start with the setup for development.
